Error on TypeScript build: standard modules has no default export

Hi,

I'm currently building a TypeScript project using `googleapis` library (version `32.0.0`, with `google-auth-library` version `1.6.0`). Whenever I try to build my project, here are the errors I get:

```bash
node_modules/google-auth-library/build/src/auth/googleauth.d.ts(18,8): error TS1192: Module '"fs"' has no default export.
node_modules/google-auth-library/build/src/auth/googleauth.d.ts(19,8): error TS1192: Module '"http"' has no default export.
node_modules/google-auth-library/build/src/auth/googleauth.d.ts(20,8): error TS1192: Module '"stream"' has no default export.
node_modules/google-auth-library/build/src/auth/jwtaccess.d.ts(17,8): error TS1192: Module '"stream"' has no default export.
node_modules/google-auth-library/build/src/auth/jwtclient.d.ts(18,8): error TS1192: Module '"stream"' has no default export.
node_modules/google-auth-library/build/src/auth/oauth2client.d.ts(18,8): error TS1192: Module '"http"' has no default export.
node_modules/google-auth-library/build/src/auth/refreshclient.d.ts(17,8): error TS1192: Module '"stream"' has no default export.
```

It seems that TypeScript expects the `import * as http from 'http'` syntax for the standard modules.

I tried with both TypeScript **2.8** and **2.9**, same issue. Node version: **8.11**.

Could you look into this? Is there something wrong with my build configuration?

Best regards,

François Voron

--- a/src/auth/googleauth.ts
+++ b/src/auth/googleauth.ts
@@ -17,12 +17,12 @@
 import {AxiosError, AxiosRequestConfig, AxiosResponse} from 'axios';
 import {exec} from 'child_process';
 import crypto from 'crypto';
-import fs from 'fs';
+import * as fs from 'fs';
 import * as gcpMetadata from 'gcp-metadata';
-import http from 'http';
+import * as http from 'http';
 import os from 'os';
 import path from 'path';
-import stream from 'stream';
+import * as stream from 'stream';
 import util from 'util';
 
 import {DefaultTransporter, Transporter} from '../transporters';

--- a/src/auth/jwtaccess.ts
+++ b/src/auth/jwtaccess.ts
@@ -16,7 +16,7 @@
 
 import jws from 'jws';
 import LRU from 'lru-cache';
-import stream from 'stream';
+import * as stream from 'stream';
 import {JWTInput} from './credentials';
 import {RequestMetadataResponse} from './oauth2client';
 

--- a/src/auth/jwtclient.ts
+++ b/src/auth/jwtclient.ts
@@ -15,7 +15,7 @@
  */
 
 import {GoogleToken} from 'gtoken';
-import stream from 'stream';
+import * as stream from 'stream';
 import {CredentialBody, Credentials, JWTInput} from './credentials';
 import {JWTAccess} from './jwtaccess';
 import {GetTokenResponse, OAuth2Client, RefreshOptions, RequestMetadataResponse} from './oauth2client';

--- a/src/auth/oauth2client.ts
+++ b/src/auth/oauth2client.ts
@@ -16,9 +16,9 @@
 
 import {AxiosError, AxiosPromise, AxiosRequestConfig, AxiosResponse} from 'axios';
 import crypto from 'crypto';
-import http from 'http';
+import * as http from 'http';
 import querystring from 'querystring';
-import stream from 'stream';
+import * as stream from 'stream';
 
 import {PemVerifier} from './../pemverifier';
 import {BodyResponseCallback} from './../transporters';

--- a/src/auth/refreshclient.ts
+++ b/src/auth/refreshclient.ts
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import stream from 'stream';
+import * as stream from 'stream';
 import {JWTInput} from './credentials';
 import {GetTokenResponse, OAuth2Client, RefreshOptions} from './oauth2client';
 

--- a/test/fixtures/kitchen/tsconfig.json
+++ b/test/fixtures/kitchen/tsconfig.json
@@ -3,9 +3,7 @@
   "compilerOptions": {
     "rootDir": ".",
     "outDir": "build",
-    // This ensures we test clients which don't have esModuleInterop
-    // enabled, which is the default.
-    "esModuleInterop": false
+    "esModuleInterop": false // test default setting
   },
   "include": [
     "src/*.ts",

--- a/test/fixtures/kitchen/tsconfig.json
+++ b/test/fixtures/kitchen/tsconfig.json
@@ -2,13 +2,13 @@
   "extends": "./node_modules/gts/tsconfig-google.json",
   "compilerOptions": {
     "rootDir": ".",
-    "outDir": "build"
+    "outDir": "build",
+    // This ensures we test clients which don't have esModuleInterop
+    // enabled, which is the default.
+    "esModuleInterop": false
   },
   "include": [
     "src/*.ts",
     "src/**/*.ts"
-  ],
-  "exclude": [
-    "node_modules"
   ]
 }

--- a/test/test.googleauth.ts
+++ b/test/test.googleauth.ts
@@ -17,12 +17,12 @@
 import assert from 'assert';
 import child_process from 'child_process';
 import crypto from 'crypto';
-import fs from 'fs';
+import * as fs from 'fs';
 import {BASE_PATH, HEADER_NAME, HOST_ADDRESS} from 'gcp-metadata';
 import nock from 'nock';
 import path from 'path';
 import sinon from 'sinon';
-import stream from 'stream';
+import * as stream from 'stream';
 
 import {GoogleAuth, GoogleAuthOptions, JWT, UserRefreshClient} from '../src';
 import {CredentialBody} from '../src/auth/credentials';

--- a/test/test.jwt.ts
+++ b/test/test.jwt.ts
@@ -15,7 +15,7 @@
  */
 
 import assert from 'assert';
-import fs from 'fs';
+import * as fs from 'fs';
 import jws from 'jws';
 import nock from 'nock';
 

--- a/test/test.jwtaccess.ts
+++ b/test/test.jwtaccess.ts
@@ -15,7 +15,7 @@
  */
 
 import assert from 'assert';
-import fs from 'fs';
+import * as fs from 'fs';
 import jws from 'jws';
 import {JWTAccess} from '../src';
 

--- a/test/test.kitchen.ts
+++ b/test/test.kitchen.ts
@@ -15,7 +15,7 @@
  */
 
 import cp from 'child_process';
-import fs from 'fs';
+import * as fs from 'fs';
 import mv from 'mv';
 import {ncp} from 'ncp';
 import pify from 'pify';

--- a/test/test.oauth2.ts
+++ b/test/test.oauth2.ts
@@ -17,7 +17,7 @@
 import assert from 'assert';
 import {AxiosError} from 'axios';
 import crypto from 'crypto';
-import fs from 'fs';
+import * as fs from 'fs';
 import nock from 'nock';
 import path from 'path';
 import qs from 'querystring';

--- a/test/test.refresh.ts
+++ b/test/test.refresh.ts
@@ -15,7 +15,7 @@
  */
 
 import assert from 'assert';
-import fs from 'fs';
+import * as fs from 'fs';
 import {UserRefreshClient} from '../src';
 
 // Creates a standard JSON credentials object for testing.

